### DISCUSSION TITLE
How to use projections with links in beanie ?

### DISCUSSION THREAD

**Discussion by aahnik** [BeanieODM/beanie#505] [Question]
```python

class UserProfile(Document):
    user: Link[UserInDB]
    owner_of_classrooms: List[Link[Classroom]] = []
  
class ClassroomIn(BaseModel):
    name: str = Field(default="New Untitled Classroom", max_length=200, min_length=3)
    description: str = Field(default=None, max_length=1000)


class ClassroomOut(ClassroomIn):
    owner_user: Link[UserInDB]
    # we store the id of the user who is the owner of this classroom
    instructors: List[Link[UserInDB]] = []
    # store list of user ids of users who are instructors
    students: List[Link[UserInDB]] = []


class Classroom(Document, ClassroomOut):
    created_time: datetime
    student_join_code: str | None = None
    instructor_join_code: str | None = None


```

Above are some code snippets from my models.

i have the user id, and I fetch the user profile using that. now I want to find the list of all classrooms that the user owns.

```python
user_profile = await UserProfile.find_one(UserProfile.user.id == current_user.id)
await user_profile.fetch_all_links()
return user_profile.owner_of_classrooms
# that would return list of Classroom

```
challenge 1 : I want to project all the Classroom in the list to ClassroomOut. 

challenge 2: and I also want to project the owner_user, instructors, students inside ClassroomOut to another thing, instead of returning full UserInDB object.

please suggest most efficient and pythonic way to achieve my objective

**Comment by roman-right** [comment]
Hey! Sorry for the delay.
Could you please show, what did you try? I could correct it if I'll see something wrong or inefficient

**Reply by aahnik** [reply]
My project is work in progress and open source. A review on my usage of mongodb and beanie is the best thing on earth 🤩

https://github.com/avabodha-space/backend-api/tree/main/api

Thanks for your help 🙏🙏

What i tried can be found in the models and routers directories. You may inspect classroom.py in both the directories. Critical feedback about how can I use beanie optimally is most appreciated. Thanks again for your time.
